### PR TITLE
6347: Evaluating parameter for ValidationUtils.validateTrue() is expensive

### PR DESCRIPTION
**Pull request by OlegMazurov**
Created variants of `ValidationUtils.validateTrue()/validateFalse()` with deferred parameter evaluation.

### CODE REVIEW

**Review by tinker-michaelj — APPROVED** [review]
LGTM, though I wonder if we even need to construct the error message `string` in these `Token` cases....it's probably not even logged at `TRACE` anywhere. 🤔

**Review by david-bakin-sl — APPROVED** [review]
Good idea! The light lambda syntax `() ->` combined with reasonable (though minimal) and inexpensive capture rules makes these deferred message-formatting thunks easy to do and appropriate for most uses.

LGTM.

File: hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/store/models/Token.java [hunk 1]
```diff
@@ -364,7 +365,7 @@ private void changeSupply(
         validateTrue(
                 treasuryRel.hasInvolvedIds(id, treasury.getId()),
                 FAIL_INVALID,
-                "Cannot change " + this + " supply (" + amount + ") with non-treasury rel " + treasuryRel);
+                () -> "Cannot change " + this + " supply (" + amount + ") with non-treasury rel " + treasuryRel);
```
[the remark this answers is not in the document]
Comment (OlegMazurov) [reply hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/store/models/Token.java:368]: I'm eliminating performance impact of `String` evaluation here (which is significant for token creation TPS). Whether those messages are needed and in what style, if yes, should be considered separately.

[the remark this answers is not in the document]
Comment (david-bakin-sl) [reply hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/store/models/Token.java:368]: I get that, that's the deferral via the lambda.  I was referring to the style of building the strings from parts only, regardless of deferral or not, and how it looks to the reader.

**Pull request merged**